Orange/widgets/data/oweditdomain.py

```diff
@@ -1182,65 +1361,63 @@ def _add_category(self):
             view.edit(index)
         self.on_values_changed()
 
-    def _merge_categories(self):
+    def _reset_name_merge(self) -> None:
+        """
+        This function resets renamed and merged variables in the model.
         """
-        Merge selected categories into one.
+        view = self.values_edit
+        model = view.model()  # type: QAbstractItemModel
+        prows = [
+            QPersistentModelIndex(model.index(i, 0))
+            for i in range(model.rowCount())
+        ]
+        with disconnected(model.dataChanged, self.on_values_changed):
+            for prow in prows:
+                if prow.isValid():
+                    model.setData(
+                        QModelIndex(prow), prow.data(SourceNameRole),
+                        Qt.EditRole
+                    )
+        self.variable_changed.emit()
 
-        Popup an editable combo box for selection/edit of a new value.
+    def _merge_less_common_categories(self) -> None:
+        """
+        Merge less common categories into one with the dialog for merge
+        selection.
         """
         view = self.values_edit
         model = view.model()  # type: QAbstractItemModel
-        rows = view.selectedIndexes()  # type: List[QModelIndex]
-        if not len(rows) >= 2:
-            return  # pragma: no cover
-        first_row = rows[0]
 
-        def mapRectTo(widget, parent, rect):
-            # type: (QWidget, QWidget, QRect) -> QRect
-            return QRect(
-                widget.mapTo(parent, rect.topLeft()),
-                rect.size(),
-            )
+        selected_attributes = [ind.data() for ind in view.selectedIndexes()]
 
-        def mapRectToGlobal(widget, rect):
-            # type: (QWidget, QRect) -> QRect
-            return QRect(
-                widget.mapToGlobal(rect.topLeft()),
-                rect.size(),
-            )
-        view.scrollTo(first_row)
-        vport = view.viewport()
-        vrect = view.visualRect(first_row)
-        vrect = mapRectTo(vport, view, vrect)
-        vrect = vrect.intersected(vport.geometry())
-        vrect = mapRectToGlobal(vport, vrect)
-
-        cb = QComboBox(editable=True, insertPolicy=QComboBox.InsertAtBottom)
-        cb.setAttribute(Qt.WA_DeleteOnClose)
-        sh = QShortcut(QKeySequence(QKeySequence.Cancel), cb)
-        sh.activated.connect(cb.close)
-        cb.setParent(self, Qt.Popup)
-        cb.move(vrect.topLeft())
-
-        cb.addItems(
-            list(unique(str(row.data(Qt.EditRole)) for row in rows)))
-        prows = [QPersistentModelIndex(row) for row in rows]
-
-        def complete_merge(text):
+        dlg = GroupLessFrequentItemsDialog(
```

Comment:
Same as above - this dialog could have a more general name, like `GroupItemsDialog`.